mocks/src/mempool.rs

```diff
@@ -99,6 +99,13 @@ impl MempoolInterface for MempoolInterfaceMock {
         }
     }
 
+    async fn transaction(
+        &self,
+        _id: &Id<Transaction>,
+    ) -> Result<Option<&SignedTransaction>, Error> {
+        Ok(None)
```

Comment:
I'm not sure why this impl for the mock exists... but this doesn't look right.

Reply:
As far as I can see, this mock is only used inside the mempool tests, so for now a proper implementation isn't needed. I have added the `todo!` macro here.